app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -41,16 +54,88 @@ def get_jwt_identity
         @authentication_parameters.jwt_identity =  @jwt_configuration.jwt_identity(@authentication_parameters)
       end
 
+      def validate_user_has_access_to_webservice
+        @validate_role_can_access_webservice.(
+          webservice: webservice,
+            account: account,
+            user_id: @authentication_parameters.jwt_identity,
+            privilege: 'authenticate'
+        )
+      end
+
+      def validate_origin
+        @validate_origin.(
+          account: account,
+          username: @authentication_parameters.jwt_identity,
+          client_ip: client_ip
+        )
+      end
+
       def validate_restrictions
         @logger.debug(LogMessages::Authentication::AuthnJwt::CALLING_VALIDATE_RESTRICTIONS.new)
         @jwt_configuration.validate_restrictions(@authentication_parameters)
       end
 
+      def audit_success
+        @audit_log.log(
+          ::Audit::Event::Authn::Authenticate.new(
+            authenticator_name: authenticator_name,
+            service: webservice,
+            role_id: audit_role_id,
+            client_ip: client_ip,
+            success: true,
+            error_message: nil
+          )
+        )
+      end
+
+      def audit_failure(err)
+        @audit_log.log(
+          ::Audit::Event::Authn::Authenticate.new(
+            authenticator_name: authenticator_name,
+            service: webservice,
+            role_id: audit_role_id,
+            client_ip: client_ip,
+            success: false,
+            error_message: err.message
+          )
+        )
+      end
+
+      def role
+        if @authentication_parameters and @authentication_parameters.jwt_identity
+          return @role_class.by_login(
+            @authentication_parameters.jwt_identity,
+            account: account
+          )
+        end
+        @role_class.by_login(
+          IDENTITY_NOT_RETRIEVED_YET,
+          account: account
+        )
+      end
+
+      def audit_role_id
+        ::Audit::Event::Authn::RoleId.new(
+          role: role,
+          account: account,
+          username: @authentication_parameters.jwt_identity
```

Comment:
Looks a bit strange. `role` already has username in it...

Reply:
This what happens also here
https://github.com/cyberark/conjur/blob/f0e9a252b13a0a6af95d50d8700d0d3dfc947f02/app/domain/authentication/authenticate.rb#L102